### PR TITLE
279 limited display time for sound mute icon

### PR DESCRIPTION
**Pull request by prakash-bhatt-rs**
#278 fix

### CODE REVIEW

File: ssr/src/page/post_view/mod.rs [hunk 1]
```diff
@@ -61,7 +61,18 @@ pub fn ScrollingView<NV: Fn() -> NVR + Clone + 'static, NVR>(
...
+        show_mute_icon,
+    } = expect_context();
+
+    set_timeout(
+        move || {
+            show_mute_icon.set(false);
+        },
+        Duration::from_secs(6),
+    );
+
```
Comment (ravibazz) [inline ssr/src/page/post_view/mod.rs:75]: should we extract out this logic into separate component? We are duplicating this code everywhere

Comment (prakash-bhatt-rs) [reply ssr/src/page/post_view/mod.rs]: Yes, we should. Pushing the update.

File: ssr/src/page/post_view/mod.rs [hunk 2]
```diff
@@ -137,7 +140,7 @@ pub fn ScrollingView<NV: Fn() -> NVR + Clone + 'static, NVR>(
                     </div>
                 </Show>
 
-                <Show when=muted>
+                <Show when=show_mute_icon>
```
Comment (ravibazz) [inline ssr/src/page/post_view/mod.rs:143]: If we click and `muted` is set to `false` but `show_mute_icon` will be set to false after 6 secs, which shouldn't happen right?.

Comment (ravibazz) [reply ssr/src/page/post_view/mod.rs:143]: Once we mute it again it it should the mute icon again for 6 secs? In this implementation that wouldn't happen. @Natasha-GB could you confirm this behaviour

**Review by ravibazz — CHANGES_REQUESTED** [review]

**Comment by Natasha-GB** [comment]
**Organic Flow**
User lands on platform > Video muted by default > Sees icon flashing for 6 secs > Can choose to unmute > Icon no longer shown if unmuted/after 6 secs

**User Journeys**
1. If user unmutes > Goes to a different page and comes back to feed > Their sound preference should be last chosen - in this case unmuted (test both icon and audio)

2. User chooses to mute again > Show flashing icon for 6 secs to denote that videos have been muted (in case user has muted unintentionally, this will help them understand why there’s no audio all of a sudden) > User can choose to unmute or continue watching muted videos

@prakash-bhatt-rs @ravibazz

**Comment by ravibazz** [comment]
> **Organic Flow** User lands on platform > Video muted by default > Sees icon flashing for 6 secs > Can choose to unmute > Icon no longer shown if unmuted/after 6 secs
> 
> **User Journeys**
> 
>     1. If user unmutes > Goes to a different page and comes back to feed > Their sound preference should be last chosen - in this case unmuted (test both icon and audio)
> 
>     2. User chooses to mute again > Show flashing icon for 6 secs to denote that videos have been muted (in case user has muted unintentionally, this will help them understand why there’s no audio all of a sudden) > User can choose to unmute or continue watching muted videos
> 
> 
> @prakash-bhatt-rs @ravibazz

Was this tested in staging? The app is not behaving that way.

**Comment by prakash-bhatt-rs** [comment]
@ravibazz No, only case tested was that sound icon stops flashing after 6seconds. Thanks

**Comment by Natasha-GB** [comment]
@prakash-bhatt-rs to fix the issue - @Natasha-GB to create a separate task for this

File: ssr/src/state/audio_state.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            Duration::from_secs(6),
+        );
+    }
+
+    pub fn toggle_mute() {
+        let Self {
+            muted,
+            show_mute_icon,
+        } = expect_context();
+        println!("Toggle Mute");
```
Comment (ravibazz) [inline ssr/src/state/audio_state.rs:56]: Can we remove this print statement

**Review by ravibazz — DISMISSED** [review]
Looks good. Just remove the debug statements

File: ssr/src/state/audio_state.rs (hunk 3 above)
Comment (prakash-bhatt-rs) [reply ssr/src/state/audio_state.rs:56]: Yes, resolving

**Comment by Natasha-GB** [comment]
Validated from product's end @prakash-bhatt-rs 

- 6 secs flashing icon checked
- Scrolling to next video - no icon checked
- Visiting different pages and back to home feed - no icon checked

Upon unmuting:
- Tap to unmute - checked
- Different page and back to home feed, preference saved - checked

Muting again:
- Tap to mute worked fine
- Icon flashed again
- Preferences worked fine

Link used: https://pr-300-go-bazzinga-hot-or-not-web-leptos-ssr.fly.dev/

**Comment by rupansh-gob** [comment]
@Natasha-GB is this ready to merge?

**Review by ravibazz — APPROVED** [review]

**Pull request merged**